scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -1105,18 +1076,26 @@ static __always_inline void p2dq_dispatch_impl(s32 cpu, struct task_struct *prev
 	u64 min_vtime = llcx->vtime;
 
 	bpf_for(i, 0, nr_dsqs_per_llc) {
-		if (llcx->dsq_max_vtime[i] < min_vtime) {
-			min_vtime = llcx->dsq_max_vtime[i];
-			dsq_id = llcx->dsqs[i];
+		cur_dsq_id = llcx->dsqs[i];
+		bpf_for_each(scx_dsq, p, cur_dsq_id, 0) {
+			if (p->scx.dsq_vtime < min_vtime) {
+				min_vtime = p->scx.dsq_vtime;
+				dsq_id = cur_dsq_id;
+			}
+			break;
```

Comment:
This seems to be a common-ish usage. We probably should add `scx_bpf_dsq_peek()` so that we can avoid the two locking cycles with the iterator. It should be possible to do this locklessly, I think.

Reply:
Created https://github.com/sched-ext/scx/issues/2184.